From 0c85ca90cb7ddb26cf85091613ca069d17fbc372 Mon Sep 17 00:00:00 2001
From: Nathan Muggli <nmuggli@google.com>
Date: Wed, 1 Jul 2026 14:00:06 -0600
Subject: [PATCH] Revert "[tracing] Migrate a few candidates to TRACE_STATE"

This reverts commit f39e44155690383376e85309dc21e2f5a04f2fa6.

Revert this commit until we get a new version of perfetto on ChromeOS.
---
 base/power_monitor/power_monitor.cc | 25 ++++++++++++++++++-------
 base/power_monitor/power_monitor.h  |  4 ++--
 2 files changed, 20 insertions(+), 9 deletions(-)

diff --git a/base/power_monitor/power_monitor.cc b/base/power_monitor/power_monitor.cc
index c31723a31c..7e62ab7407 100644
--- a/base/power_monitor/power_monitor.cc
+++ b/base/power_monitor/power_monitor.cc
@@ -179,8 +179,10 @@ void PowerMonitor::NotifyPowerStateChange(
                      : "Off")
              << " battery";
     if (emit_global_event_) {
-      TRACE_STATE("base.power", BatteryStatusToString(battery_power_status),
-                  battery_power_track_);
+      TRACE_EVENT_END("base.power", battery_power_track_);
+      TRACE_EVENT_BEGIN("base.power",
+                        BatteryStatusToString(battery_power_status),
+                        battery_power_track_);
     }
   }
 
@@ -198,12 +200,15 @@ void PowerMonitor::NotifySuspend() {
   DVLOG(1) << "Power Suspending";
   if (emit_global_event_) {
     base::trace_event::EmitNamedTrigger("power-suspend");
+    TRACE_EVENT_INSTANT("base.power", "PowerMonitor::NotifySuspend",
+                        suspend_track_);
   }
 
   AutoLock auto_lock(is_system_suspended_lock_);
   if (!is_system_suspended_) {
     if (emit_global_event_) {
-      TRACE_STATE("base.power", "PowerMonitor::Suspended", suspend_track_);
+      TRACE_EVENT_BEGIN("base.power", "PowerMonitor::Suspended",
+                        suspend_track_);
     }
     is_system_suspended_ = true;
     last_system_resume_time_ = TimeTicks::Max();
@@ -217,6 +222,8 @@ void PowerMonitor::NotifyResume() {
   DVLOG(1) << "Power Resuming";
   if (emit_global_event_) {
     base::trace_event::EmitNamedTrigger("power-resume");
+    TRACE_EVENT_INSTANT("base.power", "PowerMonitor::NotifyResume",
+                        suspend_track_);
   }
 
   TimeTicks resume_time = TimeTicks::Now();
@@ -224,7 +231,7 @@ void PowerMonitor::NotifyResume() {
   AutoLock auto_lock(is_system_suspended_lock_);
   if (is_system_suspended_) {
     if (emit_global_event_) {
-      TRACE_STATE("base.power", nullptr, suspend_track_);
+      TRACE_EVENT_END("base.power", suspend_track_);
     }
     is_system_suspended_ = false;
     last_system_resume_time_ = resume_time;
@@ -284,13 +291,17 @@ void PowerMonitor::OnStart(const perfetto::DataSourceBase::StartArgs&) {
   {
     AutoLock auto_lock(is_system_suspended_lock_);
     if (is_system_suspended_) {
-      TRACE_STATE("base.power", "PowerMonitor::Suspended", suspend_track_);
+      TRACE_EVENT_END("base.power", suspend_track_);
+      TRACE_EVENT_BEGIN("base.power", "PowerMonitor::Suspended",
+                        suspend_track_);
     }
   }
   {
     AutoLock auto_lock(battery_power_status_lock_);
-    TRACE_STATE("base.power", BatteryStatusToString(battery_power_status_),
-                battery_power_track_);
+    TRACE_EVENT_END("base.power", battery_power_track_);
+    TRACE_EVENT_BEGIN("base.power",
+                      BatteryStatusToString(battery_power_status_),
+                      battery_power_track_);
   }
 }
 
diff --git a/base/power_monitor/power_monitor.h b/base/power_monitor/power_monitor.h
index 32db015088..4ad002654c 100644
--- a/base/power_monitor/power_monitor.h
+++ b/base/power_monitor/power_monitor.h
@@ -137,8 +137,8 @@ class BASE_EXPORT PowerMonitor : public perfetto::TrackEventSessionObserver {
   mutable Lock battery_power_status_lock_;
 
   bool emit_global_event_ = false;
-  const perfetto::StateTrack suspend_track_;
-  const perfetto::StateTrack battery_power_track_;
+  const perfetto::NamedTrack suspend_track_;
+  const perfetto::NamedTrack battery_power_track_;
   PowerThermalObserver::DeviceThermalState power_thermal_state_
       GUARDED_BY(power_thermal_state_lock_) =
           PowerThermalObserver::DeviceThermalState::kUnknown;
-- 
2.55.0.rc0.799.gd6f94ed593-goog

